src/events/guilds/members/rawMemberRemoveNotify.ts

```diff
@@ -0,0 +1,79 @@
+import { GuildSettings } from '@lib/database';
+import { Colors } from '@lib/types/constants/Constants';
+import { Events } from '@lib/types/Enums';
+import { LanguageKeys } from '@lib/types/namespaces/LanguageKeys';
+import { ApplyOptions } from '@skyra/decorators';
+import { MessageLogsEnum, Moderation } from '@utils/constants';
+import { getDisplayAvatar } from '@utils/util';
+import { GatewayGuildMemberRemoveDispatch } from 'discord-api-types/v6';
+import { Guild, GuildMember, MessageEmbed } from 'discord.js';
+import { Event, EventOptions } from 'klasa';
+
+@ApplyOptions<EventOptions>({ name: Events.RawMemberRemove })
+export default class extends Event {
+	public async run(guild: Guild, data: GatewayGuildMemberRemoveDispatch['d']) {
+		const [enabled, language] = await guild.readSettings((settings) => [settings[GuildSettings.Events.MemberRemove], settings.getLanguage()]);
+		if (!enabled) return;
+
+		const member = guild.members.cache.get(data.user.id);
+		const isModerationAction = await this.isModerationAction(guild, data);
+
+		const footer = isModerationAction.kicked
+			? language.get(LanguageKeys.Events.GuildMemberKicked)
+			: isModerationAction.banned
+			? language.get(LanguageKeys.Events.GuildMemberBanned)
+			: isModerationAction.softbanned
+			? language.get(LanguageKeys.Events.GuildMemberSoftBanned)
+			: language.get(LanguageKeys.Events.GuildMemberRemove);
+
+		const time = this.processJoinedTimestamp(member);
+		this.client.emit(Events.GuildMessageLog, MessageLogsEnum.Member, guild, () =>
+			new MessageEmbed()
+				.setColor(Colors.Red)
+				.setAuthor(`${data.user.username}#${data.user.discriminator} (${data.user.id})`, getDisplayAvatar(data.user.id, data.user))
```

Comment:
Maybe you should look into `client.users.add(data.user, false)` and then using user.displayAvatarURL / user.tag?

Reply:
No, that'd needlessly duplicate ops.